src/classes/BGE_DataImportBatchEntry_CTRL.cls

```diff
@@ -322,72 +383,62 @@ public with sharing class BGE_DataImportBatchEntry_CTRL {
     }
 
     /*******************************************************************************************************
-    * @description runs BDI Dry Run against a list of DataImport__c Ids
-    * based on the processing settings of a given batch
-    * @return the full data set for the batch
+    * @description prepare the data import record for dry run by clearing certain fields or retaining them
+    * based on if user has already selected a match, or if they want to let matching run. Either way, we want to use Dry Run
+    * to check for potential errors.
+    * @return List<DataImport__c> because BDI expects a list
```

Comment:
We could, but then it's just really easy to pass directly to the processDataImportRecords. 

Since I named this 'prepareDataImportForDryRun', I feel this is an acceptable weird thing, because I'd rather make it strange here than change that BDI method. 

That said, if we think this is really weird, then I'd make another BDI_DataImport_API method like 'processSingleDataImportRecord' and remove the list processing there. Thoughts?